test/multisigwallet.test.ts

```diff
@@ -0,0 +1,521 @@
+
+    beforeEach(async () => {
+      const tx = await submitAddOwner({ newOwner: accounts[9] });
+      transactionId = findLastTransactionId(tx);
+      await wallet.confirmTransaction(transactionId, { from: owners[1] });
+    });
+
+    it('should revert for non-owner', async () => {
+      await assertReverts(async () => {
+        await wallet.executeTransaction(transactionId, { from: accounts[9] });
+      });
+    });
+
+    it('should revert for non-existing transaction', async () => {
+      await assertReverts(async () => {
+        await wallet.executeTransaction(2, { from: owners[0] });
+      });
+    });
+
+    it('should revert for not confirmed transaction', async () => {
+      await assertReverts(async () => {
+        await wallet.executeTransaction(transactionId, { from: owners[2] });
+      });
+    });
+
+    it('should emit ExecutionEvent', async () => {
+      const tx = await wallet.confirmTransaction(transactionId, {
+        from: owners[2]
+      });
+
+      const log = findLastLog(tx, 'Execution');
+      assert.isOk(log);
+
+      const event = log.args as ExecutionEvent;
+      assert.isOk(event);
+    });
+
+    it('should set transaction as executed', async () => {
+      await wallet.confirmTransaction(transactionId, {
+        from: owners[2]
+      });
+
+      const transactionCount = await wallet.getTransactionCount(false, true);
+      assert.isTrue(new BigNumber(1).eq(transactionCount));
+
+      const transactionIds = await wallet.getTransactionIds(0, 1, false, true);
+      assert.isTrue(transactionId.eq(transactionIds[0]));
+    });
+  });
+
+  describe('#changeRequirement', () => {
+    const newRequired = new BigNumber(2);
+
+    it('should revert for not wallet', async () => {
+      await assertReverts(async () => {
+        await wallet.changeRequirement(newRequired);
+      });
+    });
+
+    it('should emit ExecutionFailureEvent for invalid requirements', async () => {
+      let tx: TransactionResult;
+      let log: TransactionLog;
+      let event: ExecutionFailureEvent;
+      tx = await executeChangeRequirements(new BigNumber(0));
+
+      log = findLastLog(tx, 'ExecutionFailure');
+      assert.isOk(log);
+      event = log.args;
+      assert.isOk(event);
+
+      tx = await executeChangeRequirements(new BigNumber(4));
+
+      log = findLastLog(tx, 'ExecutionFailure');
+      assert.isOk(log);
+      event = log.args;
+      assert.isOk(event);
+    });
+
+    it('should emit RequirementChangeEvent', async () => {
+      const tx = await executeChangeRequirements(new BigNumber(newRequired));
+
+      const log = findLastLog(tx, 'RequirementChange');
+      assert.isOk(log);
+      const event = log.args as RequirementChangeEvent;
+      assert.isOk(event);
+    });
+
+    it('should change `required` attribute', async () => {
+      const tx = await executeChangeRequirements(new BigNumber(newRequired));
+
+      const walletRequired = findNewRequirement(tx);
```

Comment:
You should check `required` against getter not event.

Reply:
This way you duplicate the previous test case 'should emit RequirementChange event'